server/sonar-db-dao/src/main/java/org/sonar/db/rule/RuleDto.java

```diff
@@ -19,367 +19,379 @@
-  public Date getNoteCreatedAt() {
-    return noteCreatedAt;
+  public String getDefRemediationBaseEffort() {
+    return definition.getDefRemediationBaseEffort();
   }
 
-  public RuleDto setNoteCreatedAt(@Nullable Date noteCreatedAt) {
-    this.noteCreatedAt = noteCreatedAt;
+  public RuleDto setDefRemediationBaseEffort(@Nullable String defaultRemediationBaseEffort) {
+    definition.setDefRemediationBaseEffort(defaultRemediationBaseEffort);
     return this;
   }
 
   @CheckForNull
-  public Date getNoteUpdatedAt() {
-    return noteUpdatedAt;
+  public String getGapDescription() {
+    return definition.getGapDescription();
   }
 
-  public RuleDto setNoteUpdatedAt(@Nullable Date noteUpdatedAt) {
-    this.noteUpdatedAt = noteUpdatedAt;
+  public RuleDto setGapDescription(@Nullable String s) {
+    definition.setGapDescription(s);
     return this;
   }
 
-  @CheckForNull
-  public String getRemediationFunction() {
-    return remediationFunction;
+  public RuleDto setSystemTags(Set<String> tags) {
+    this.definition.setSystemTags(tags);
+    return this;
   }
 
-  public RuleDto setRemediationFunction(@Nullable String remediationFunction) {
-    this.remediationFunction = remediationFunction;
-    return this;
+  public int getType() {
+    return definition.getType();
   }
 
-  @CheckForNull
-  public String getDefaultRemediationFunction() {
-    return defRemediationFunction;
+  public RuleDto setType(int type) {
+    definition.setType(type);
+    return this;
   }
 
-  public RuleDto setDefaultRemediationFunction(@Nullable String defaultRemediationFunction) {
-    this.defRemediationFunction = defaultRemediationFunction;
+  public RuleDto setType(RuleType type) {
+    definition.setType(type);
     return this;
   }
 
-  @CheckForNull
-  public String getRemediationGapMultiplier() {
-    return remediationGapMultiplier;
+  public Set<String> getSystemTags() {
+    return definition.getSystemTags();
   }
 
-  public RuleDto setRemediationGapMultiplier(@Nullable String remediationGapMultiplier) {
-    this.remediationGapMultiplier = remediationGapMultiplier;
-    return this;
+  /**
+   * Used in MyBatis mapping.
+   */
+  private void setSystemTagsField(String s) {
+    definition.setSystemTagsField(s);
   }
 
-  @CheckForNull
-  public String getDefaultRemediationGapMultiplier() {
-    return defRemediationGapMultiplier;
+  public long getCreatedAt() {
+    return definition.getCreatedAt();
   }
 
-  public RuleDto setDefaultRemediationGapMultiplier(@Nullable String defaultRemediationGapMultiplier) {
-    this.defRemediationGapMultiplier = defaultRemediationGapMultiplier;
+  public RuleDto setCreatedAt(long createdAt) {
+    definition.setCreatedAt(createdAt);
+    metadata.setCreatedAt(createdAt);
     return this;
   }
 
-  @CheckForNull
-  public String getRemediationBaseEffort() {
-    return remediationBaseEffort;
+  /**
+   * Used in MyBatis mapping.
+   */
+  private void setCreatedAtFromDefinition(@Nullable Long createdAt) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused private "setCreatedAtFromDefinition" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AUnusedPrivateMethod)